### PR TITLE
MH3 20 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/aether_spike.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Name:Aether Spike
+ManaCost:1 U
+Types:Instant
+A:SP$ Pump | TargetType$ Spell | TgtZone$ Stack | ValidTgts$ Card | SubAbility$ DBPutCounter | StackDescription$ {p:You} chooses {s:Targeted}. | SpellDescription$ Choose target spell.
+SVar:DBPutCounter:DB$ PutCounter | Defined$ You | AILogic$ PayEnergy | CounterType$ ENERGY | CounterNum$ 2 | SubAbility$ DBChooseNumber | SpellDescription$ You get {E}{E} (two energy counters), then you may pay any amount of {E}. 
+SVar:DBChooseNumber:DB$ ChooseNumber | Max$ Max | ListTitle$ Pay Energy to counter targeted spell | SubAbility$ DBPay
+SVar:DBPay:DB$ Pump | UnlessCost$ PayEnergy<X> | UnlessPayer$ You | UnlessSwitched$ True | SubAbility$ DBCounter
+SVar:DBCounter:DB$ Counter | Defined$ Targeted | UnlessCost$ X | UnlessPayer$ TargetedController | SpellDescription$ Counter that spell unless its controller pays [1} for each {E} paid this way.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aether_spike.txt:8]: that's not a X shard
use a different char

File: forge-gui/res/cardsfolder/upcoming/aether_spike.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Name:Aether Spike
+ManaCost:1 U
+Types:Instant
+A:SP$ Pump | TargetType$ Spell | TgtZone$ Stack | ValidTgts$ Card | SubAbility$ DBPutCounter | StackDescription$ {p:You} chooses {s:Targeted}. | SpellDescription$ Choose target spell.
+SVar:DBPutCounter:DB$ PutCounter | Defined$ You | AILogic$ PayEnergy | CounterType$ ENERGY | CounterNum$ 2 | SubAbility$ DBChooseNumber | SpellDescription$ You get {E}{E} (two energy counters), then you may pay any amount of {E}. 
+SVar:DBChooseNumber:DB$ ChooseNumber | Max$ Max | ListTitle$ Pay Energy to counter targeted spell | SubAbility$ DBPay
+SVar:DBPay:DB$ Pump | UnlessCost$ PayEnergy<X> | UnlessPayer$ You | UnlessSwitched$ True | SubAbility$ DBCounter
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aether_spike.txt:7]: what happens when you don't pay?

File: forge-gui/res/cardsfolder/upcoming/ajani_fells_the_godsire.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:Ajani Fells the Godsire
+ManaCost:3 W W
+Types:Enchantment Saga
+K:Chapter:3:DBExile,DBCat,DBPump
+SVar:DBExile:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | ValidTgts$ Permanent.OppCtrl+powerGE3 | TgtPrompt$ Select target permanent an opponent controls with power 3 or greater | SpellDescription$ Exile target permanent an opponent controls with power 3 or greater.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ajani_fells_the_godsire.txt:5]: wrong tgts?

File: forge-gui/res/cardsfolder/upcoming/archway_of_innovation.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:Archway of Innovation
+ManaCost:no cost
+Types:Land
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionPresent$ Island.YouCtrl | ConditionCompare$ EQ0 | SpellDescription$ CARDNAME enters the battlefield tapped unless you control an Island.
+A:AB$ Mana | Cost$ T | Produced$ U | SpellDescription$ Add {U}.
+A:AB$ Effect | Cost$ UT | StaticAbilities$ GrantImprovise | Triggers$ ExileEffect | SpellDescription$ The next spell you cast this turn has improvise.(Your artifacts can help cast this spell. Each artifact you tap after you're done activating mana abilities pays for {1}.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/archway_of_innovation.txt:7]: bad cost

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/archway_of_innovation.txt:7]: spacing

File: forge-gui/res/cardsfolder/upcoming/boggart_trawler_boggart_bog.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Name:Boggart Trawler 
+ManaCost:2 B
+Types:Creature Goblin
+PT:3/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ When CARDNAME enters the battlefield, exile all cards from target player's graveyard.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/boggart_trawler_boggart_bog.txt:5]: wrong description...

File: forge-gui/res/cardsfolder/upcoming/boggart_trawler_boggart_bog.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+ALTERNATE
+
+Name:Boggart Bog
+ManaCost:no cost
+Types:Land
+K:ETBReplacement:Other:DBTap
+SVar:DBTap:DB$ Tap | ETB$ True | Defined$ Self | UnlessCost$ PayLife<3> | UnlessPayer$ You | StackDescription$ enters the battlefield tapped. | SpellDescription$ As CARDNAME enters the battlefield, you may pay 3 life. If you don't, it enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ B | SpellDescription$ Add {B}.
+Oracle:Boggart Bog enters the battlefield, you may pay 3 life. If you don't, it enters the battlefield tapped.\n{T}: Add {B}.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/boggart_trawler_boggart_bog.txt:18]: incomplete...?

File: forge-gui/res/cardsfolder/upcoming/imskir_iron_eater.txt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Name:Imskir Iron-Eater
+ManaCost:6 B R
+Types:Legendary Creature Demon
+PT:5/5
+K:Affinity:Artifact
+T:Mode$ ChangesZone | ValidCard$ Card.Self | Destination$ Battlefield | Execute$ TrigDraw | TriggerDescription$ When CARDNAME enters the battlefield, you draw X cards and you lose X life, where X is half the number of artifacts you control, rounded down.
+SVar:TrigDraw:DB$ Draw | NumCards$ X | SubAbility$ DBLoseLife
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/imskir_iron_eater.txt:7]: wrong Num

File: forge-gui/res/cardsfolder/upcoming/phelia_exuberant_shepherd.txt [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Name:Phelia, Exuberant Shepherd
+ManaCost:1 W
+Types:Legendary Creature Dog
+PT:2/2
+K:Flash
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever CARDNAME attacks, exile up to one other target nonland permanent. At the beginning of the next end step, return that card to the battlefield under its owner's control. If it entered under your control, put a +1/+1 counter on NICKNAME.
+SVar:TrigExile:DB$ ChangeZone | ValidTgts$ Permanent.Other | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one other target permanent | Origin$ Battlefield | Destination$ Exile | RememberChanged$ True | SubAbility$ DelTrig
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigBounce | TriggerDescription$ Return exiled permanent to the battlefield. | RememberObjects$ RememberedLKI | SubAbility$ DBCleanup
+SVar:TrigBounce:DB$ ChangeZone | Origin$ Exile | Destination$ Battlefield | Defined$ DelayTriggerRememberedLKI | SubAbility$ DBPutCounter
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | ConditionDefined$ DelayTriggerRememberedLKI | ConditionPresent$ Creature
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/phelia_exuberant_shepherd.txt:10]: wrong condition

File: forge-gui/res/cardsfolder/upcoming/the_necrobloom.txt [hunk 9]
```diff
@@ -0,0 +1,13 @@
+Name:The Necrobloom
+ManaCost:1 W B G
+Types:Legendary Creature Plant
+PT:2/7
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Land.YouCtrl | Execute$ TrigBranch | TriggerZones$ Battlefield | TriggerDescription$ Landfall – Whenever a land enters the battlefield under your control, create a 0/1 green Plant creature token. If you control seven or more lands with different names, create a 2/2 black Zombie creature token instead.
+SVar:TrigBranch:DB$ Branch | BranchConditionSVar$ X | BranchConditionSVarCompare$ GE6 | TrueSubAbility$ DBZombie | FalseSubAbility$ DBPlant
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_necrobloom.txt:6]: 6?

File: forge-gui/res/cardsfolder/upcoming/powerbalance.txt [hunk 10]
```diff
@@ -0,0 +1,9 @@
+Name:Powerbalance 
+ManaCost:R R
+Types:Enchantment
+T:Mode$ SpellCast | ValidCard$ Card | ValidActivatingPlayer$ Opponent | OptionalDecider$ You | TriggerZones$ Battlefield | Execute$ TrigReveal | TriggerDescription$ Whenever an opponent casts a spell, you may reveal the top card of your library. If you do, you may cast that card without paying its mana cost if the two spells have the same mana value.
+SVar:TrigReveal:DB$ Dig | DigNum$ 1 | Reveal$ True | DestinationZone$ Library | LibraryPosition$ 0 | LibraryPosition2$ 0 | RememberRevealed$ True | SubAbility$ DBPlay
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/powerbalance.txt:5]: use PeekAndReveal

File: forge-gui/res/cardsfolder/upcoming/ripples_of_undeath.txt [hunk 11]
```diff
@@ -0,0 +1,9 @@
+Name:Ripples of Undeath
+ManaCost:1 B
+Types:Enchantment
+T:Mode$ Phase | PreCombatMain$ True | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigMill | TriggerDescription$ At the beginning of your precombat main phase, mill three cards. Then you may pay {1} and 3 life. If you do, put a card from among those cards into your hand.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ripples_of_undeath.txt:4]: better use `Phase$ Main1`

File: forge-gui/res/cardsfolder/upcoming/aether_spike.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/aether_spike.txt]: I don't think I understand the problem ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/aether_spike.txt]: Did you test your changes?
I have doubts since you introduced xPaid now...

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/aether_spike.txt]: does it work if you only use N for everything?

File: forge-gui/res/cardsfolder/upcoming/rosheen_roaring_prophet.txt [hunk 12]
```diff
@@ -0,0 +1,13 @@
+Name:Rosheen, Roaring Prophet 
+ManaCost:2 R G
+Types:Legendary Creature Giant Shaman
+PT:4/4
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMill | TriggerDescription$ When CARDNAME enters the battlefield, mill six cards. You may put a card with X in its mana cost from among them into your hand.
+SVar:TrigMill:DB$ Mill | Defined$ You | NumCards$ 6 | RememberMilled$ True | SubAbility$ DBChangeZone
+SVar:DBChangeZone:DB$ ChangeZone | Hidden$ True | Origin$ Graveyard,Exile | Destination$ Hand | ChangeType$ Card.IsRemembered+hasXCost | SelectPrompt$ You may put a card with X in its mana cost from among them into your hand
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rosheen_roaring_prophet.txt:7]: cleanup missing

File: forge-gui/res/cardsfolder/upcoming/rosheen_roaring_prophet.txt [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+A:AB$ Reveal | Cost$ T | RevealValid$ Card.hasXCost+YouCtrl | AnyNumber$ True | RememberRevealed$ True | SubAbility$ DBMana | SpellDescription$ Reveal any number of cards with X in their mana cost in your hand. Add {C}{C} for each card revealed this way. Spend this mana only on costs that contain X.
+SVar:DBMana:DB$ Mana | Produced$ C | Amount$ MetalWorkerX | RestrictValid$ Spell.hasXCost | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rosheen_roaring_prophet.txt:9]: wrong restriction

File: forge-gui/res/cardsfolder/upcoming/shilgengar_sire_of_famine.txt [hunk 14]
```diff
@@ -0,0 +1,14 @@
+Name:Shilgengar, Sire of Famine
+ManaCost:3 B B
+Types:Legendary Creature Elder Demon
+PT:6/6
+K:Flying
+A:AB$ Token | Cost$ Sac<1/Creature.Other/another creature> | TokenScript$ c_a_blood_draw | TokenAmount$ X | SpellDescription$  Create a Blood token. If you sacrificed an Angel this way, create a number of Blood tokens equal to its toughness instead.
+SVar:X:Count$Compare AngelSac EQ1.SacT.1
+SVar:SacT:Sacrificed$CardToughness
+SVar:AngelSac:Sacrificed$Valid Card.Angel
+A:AB$ ChangeZoneAll | Cost$ WB WB WB Sac<6/Blood.token/Blood token> | WithCountersType$ FINALITY | AnimateSubAbility$ DBAnimate | ChangeType$ Creature.YouOwn | Origin$ Graveyard | Destination$ Battlefield | SpellDescription$ Return each creature card from your graveyard to the battlefield with a finality counter on it. Those creatures are Vampires in addition to their other types. Return each creature card from your graveyard to the battlefield with a finality counter on it. Those creatures are Vampires in addition to their other types.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shilgengar_sire_of_famine.txt:10]: bad description

File: forge-gui/res/cardsfolder/upcoming/sink_into_stupor_soporific_springs.txt [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+ALTERNATE
+
+Name:Soporific Springs
+ManaCost:no cost
+Types:Land
+K:ETBReplacement:Other:DBTap
+SVar:DBTap:DB$ Tap | ETB$ True | Defined$ Self | UnlessCost$ PayLife<3> | UnlessPayer$ You | StackDescription$ enters the battlefield tapped. | SpellDescription$ As CARDNAME enters the battlefield, you may pay 3 life. If you don't, it enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ U | SpellDescription$ Add {U}.
+Oracle:As Soporific Springs enters the battlefield, you may pay 3 life. If you don't, it enters the battlefield tapped.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sink_into_stupor_soporific_springs.txt:16]: incomplete

File: forge-gui/res/cardsfolder/upcoming/spymasters_vault.txt [hunk 16]
```diff
@@ -0,0 +1,9 @@
+Name:Spymaster's Vault
+ManaCost:no cost
+Types:Land
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionPresent$ Swamp.YouCtrl | ConditionCompare$ EQ0 | SpellDescription$
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/spymasters_vault.txt:5]: incomplete

File: forge-gui/res/cardsfolder/upcoming/aether_spike.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/aether_spike.txt]: I think so ? 
image

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/aether_spike.txt]: now just fix that StackDescription ;)

File: forge-gui/res/cardsfolder/upcoming/phelia_exuberant_shepherd.txt (hunk 8 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/phelia_exuberant_shepherd.txt]: checking against LKI still seems wrong
imo best to imprint the zone change instead

**Review by tool4ever — APPROVED** [review]

**Pull request merged**